MozillaReality/FirefoxReality
# Follow system language

This adds support for a "Follow system language" placeholder that uses whatever the system locale is as long as it's supported by FxR, otherwise it falls back to en-US. 

Also fixes some issues with the correct language hot reload. Now we should be able to change the language either from the Settings or from the system settings and everything so be instantly localized in the new language.

Things to test:
- Change the language from the language setting: All the UI strings should be now localized in the new language
- Click home - Go to the system settings - Change the language - Go back to the app: All the UI strings should be now localized in the new language
- Reset the languages: The UI should show "Follow the system Language" and all the strings should be localized in the system language. In case of an unsupported language (ie. pt-PT), it should fall back to en-US
- Set a display language explicitly - Click home - Go to the system settings - Change the language - Go back to the app: The user chosen language should remain as the UI language.
- Check that when "Follow system language" is selected the voice search is performed in the correct language based on the system one. In case of an unsupported language (ie. pt-PT), it should fall back to en-US
- Test updating from v8. The app should retain the previous language configuration and show specific locales for all three language settings.
- All the sumo pages should be shown localized in the content language order. They were previously hardcoded to en-US.
